File: server-java-play-framework/app/controllers/AssignmentApiController.java
package controllers;

import apimodels.AccountMiniResponse;
import apimodels.AssignmentResponse;
import apimodels.AssignmentStatusResponse;
import java.math.BigDecimal;
import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class AssignmentApiController extends Controller {
    private final AssignmentApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private AssignmentApiController(Config configuration, AssignmentApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result assigmentAssigneeAccountSearch(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuekeyword = request.getQueryString("keyword");
        String keyword;
        if (valuekeyword != null) {
            keyword = valuekeyword;
        } else {
            keyword = null;
        }
        return imp.assigmentAssigneeAccountSearchHttp(request, version, accountId, keyword);
    }

    @ApiAction
    public Result assignmentCreate(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuename = request.getQueryString("name");
        String name;
        if (valuename != null) {
            name = valuename;
        } else {
            throw new IllegalArgumentException("'name' parameter is required");
        }
        String valuedescription = request.getQueryString("description");
        String description;
        if (valuedescription != null) {
            description = valuedescription;
        } else {
            description = null;
        }
        String valueassigneeAccountId = request.getQueryString("assigneeAccountId");
        Long assigneeAccountId;
        if (valueassigneeAccountId != null) {
            assigneeAccountId = Long.parseLong(valueassigneeAccountId);
        } else {
            throw new IllegalArgumentException("'assigneeAccountId' parameter is required");
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valuetags = request.getQueryString("tags");
        String tags;
        if (valuetags != null) {
            tags = valuetags;
        } else {
            tags = null;
        }
        String valueactive = request.getQueryString("active");
        Boolean active;
        if (valueactive != null) {
            active = Boolean.valueOf(valueactive);
        } else {
            active = null;
        }
        return imp.assignmentCreateHttp(request, version, accountId, name, assigneeAccountId, description, retailerLocationId, tags, active);
    }

    @ApiAction
    public Result assignmentDelete(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentId = request.getQueryString("assignmentId");
        Long assignmentId;
        if (valueassignmentId != null) {
            assignmentId = Long.parseLong(valueassignmentId);
        } else {
            throw new IllegalArgumentException("'assignmentId' parameter is required");
        }
        return imp.assignmentDeleteHttp(request, version, accountId, assignmentId);
    }

    @ApiAction
    public Result assignmentGet(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentId = request.getQueryString("assignmentId");
        Long assignmentId;
        if (valueassignmentId != null) {
            assignmentId = Long.parseLong(valueassignmentId);
        } else {
            throw new IllegalArgumentException("'assignmentId' parameter is required");
        }
        return imp.assignmentGetHttp(request, version, accountId, assignmentId);
    }

    @ApiAction
    public Result assignmentSearch(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valuecreatorAccountId = request.getQueryString("creatorAccountId");
        Long creatorAccountId;
        if (valuecreatorAccountId != null) {
            creatorAccountId = Long.parseLong(valuecreatorAccountId);
        } else {
            creatorAccountId = null;
        }
        String valueassigneeAccountIds = request.getQueryString("assigneeAccountIds");
        String assigneeAccountIds;
        if (valueassigneeAccountIds != null) {
            assigneeAccountIds = valueassigneeAccountIds;
        } else {
            assigneeAccountIds = null;
        }
        String valueretailerLocationIds = request.getQueryString("retailerLocationIds");
        String retailerLocationIds;
        if (valueretailerLocationIds != null) {
            retailerLocationIds = valueretailerLocationIds;
        } else {
            retailerLocationIds = null;
        }
        String valuecurrentStatusType = request.getQueryString("currentStatusType");
        String currentStatusType;
        if (valuecurrentStatusType != null) {
            currentStatusType = valuecurrentStatusType;
        } else {
            currentStatusType = null;
        }
        String valuekeyword = request.getQueryString("keyword");
        String keyword;
        if (valuekeyword != null) {
            keyword = valuekeyword;
        } else {
            keyword = null;
        }
        String valuesortField = request.getQueryString("sortField");
        String sortField;
        if (valuesortField != null) {
            sortField = valuesortField;
        } else {
            throw new IllegalArgumentException("'sortField' parameter is required");
        }
        String valuedescending = request.getQueryString("descending");
        Boolean descending;
        if (valuedescending != null) {
            descending = Boolean.valueOf(valuedescending);
        } else {
            throw new IllegalArgumentException("'descending' parameter is required");
        }
        String valueactiveOnly = request.getQueryString("activeOnly");
        Boolean activeOnly;
        if (valueactiveOnly != null) {
            activeOnly = Boolean.valueOf(valueactiveOnly);
        } else {
            throw new IllegalArgumentException("'activeOnly' parameter is required");
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            throw new IllegalArgumentException("'start' parameter is required");
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            throw new IllegalArgumentException("'limit' parameter is required");
        }
        return imp.assignmentSearchHttp(request, version, accountId, sortField, descending, activeOnly, start, limit, creatorAccountId, assigneeAccountIds, retailerLocationIds, currentStatusType, keyword);
    }

    @ApiAction
    public Result assignmentStatusCreate(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentId = request.getQueryString("assignmentId");
        Long assignmentId;
        if (valueassignmentId != null) {
            assignmentId = Long.parseLong(valueassignmentId);
        } else {
            throw new IllegalArgumentException("'assignmentId' parameter is required");
        }
        String valuescheduledNotificationId = request.getQueryString("scheduledNotificationId");
        Long scheduledNotificationId;
        if (valuescheduledNotificationId != null) {
            scheduledNotificationId = Long.parseLong(valuescheduledNotificationId);
        } else {
            scheduledNotificationId = null;
        }
        String valuetoDo = request.getQueryString("toDo");
        String toDo;
        if (valuetoDo != null) {
            toDo = valuetoDo;
        } else {
            toDo = null;
        }
        String valueconnection = request.getQueryString("connection");
        String connection;
        if (valueconnection != null) {
            connection = valueconnection;
        } else {
            connection = null;
        }
        String valuemethod = request.getQueryString("method");
        String method;
        if (valuemethod != null) {
            method = valuemethod;
        } else {
            method = null;
        }
        String valuestatus = request.getQueryString("status");
        String status;
        if (valuestatus != null) {
            status = valuestatus;
        } else {
            status = null;
        }
        String valueclosure = request.getQueryString("closure");
        String closure;
        if (valueclosure != null) {
            closure = valueclosure;
        } else {
            closure = null;
        }
        String valuemessage = request.getQueryString("message");
        String message;
        if (valuemessage != null) {
            message = valuemessage;
        } else {
            message = null;
        }
        String valuefollowUp = request.getQueryString("followUp");
        Long followUp;
        if (valuefollowUp != null) {
            followUp = Long.parseLong(valuefollowUp);
        } else {
            followUp = null;
        }
        String valueactive = request.getQueryString("active");
        Boolean active;
        if (valueactive != null) {
            active = Boolean.valueOf(valueactive);
        } else {
            active = null;
        }
        return imp.assignmentStatusCreateHttp(request, version, accountId, assignmentId, scheduledNotificationId, toDo, connection, method, status, closure, message, followUp, active);
    }

    @ApiAction
    public Result assignmentStatusDelete(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentStatusId = request.getQueryString("assignmentStatusId");
        Long assignmentStatusId;
        if (valueassignmentStatusId != null) {
            assignmentStatusId = Long.parseLong(valueassignmentStatusId);
        } else {
            throw new IllegalArgumentException("'assignmentStatusId' parameter is required");
        }
        return imp.assignmentStatusDeleteHttp(request, version, accountId, assignmentStatusId);
    }

    @ApiAction
    public Result assignmentStatusGet(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentStatusId = request.getQueryString("assignmentStatusId");
        Long assignmentStatusId;
        if (valueassignmentStatusId != null) {
            assignmentStatusId = Long.parseLong(valueassignmentStatusId);
        } else {
            throw new IllegalArgumentException("'assignmentStatusId' parameter is required");
        }
        return imp.assignmentStatusGetHttp(request, version, accountId, assignmentStatusId);
    }

    @ApiAction
    public Result assignmentStatusSearch(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentId = request.getQueryString("assignmentId");
        Long assignmentId;
        if (valueassignmentId != null) {
            assignmentId = Long.parseLong(valueassignmentId);
        } else {
            assignmentId = null;
        }
        String valuecreatorAccountId = request.getQueryString("creatorAccountId");
        Long creatorAccountId;
        if (valuecreatorAccountId != null) {
            creatorAccountId = Long.parseLong(valuecreatorAccountId);
        } else {
            creatorAccountId = null;
        }
        String valueassigneeAccountId = request.getQueryString("assigneeAccountId");
        Long assigneeAccountId;
        if (valueassigneeAccountId != null) {
            assigneeAccountId = Long.parseLong(valueassigneeAccountId);
        } else {
            assigneeAccountId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valuestatusType = request.getQueryString("statusType");
        String statusType;
        if (valuestatusType != null) {
            statusType = valuestatusType;
        } else {
            statusType = null;
        }
        String valuekeyword = request.getQueryString("keyword");
        String keyword;
        if (valuekeyword != null) {
            keyword = valuekeyword;
        } else {
            keyword = null;
        }
        String valuesortField = request.getQueryString("sortField");
        String sortField;
        if (valuesortField != null) {
            sortField = valuesortField;
        } else {
            throw new IllegalArgumentException("'sortField' parameter is required");
        }
        String valuedescending = request.getQueryString("descending");
        Boolean descending;
        if (valuedescending != null) {
            descending = Boolean.valueOf(valuedescending);
        } else {
            throw new IllegalArgumentException("'descending' parameter is required");
        }
        String valueactiveOnly = request.getQueryString("activeOnly");
        Boolean activeOnly;
        if (valueactiveOnly != null) {
            activeOnly = Boolean.valueOf(valueactiveOnly);
        } else {
            throw new IllegalArgumentException("'activeOnly' parameter is required");
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            throw new IllegalArgumentException("'start' parameter is required");
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            throw new IllegalArgumentException("'limit' parameter is required");
        }
        return imp.assignmentStatusSearchHttp(request, version, accountId, sortField, descending, activeOnly, start, limit, assignmentId, creatorAccountId, assigneeAccountId, retailerLocationId, statusType, keyword);
    }

    @ApiAction
    public Result assignmentStatusUpdate(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentStatusId = request.getQueryString("assignmentStatusId");
        Long assignmentStatusId;
        if (valueassignmentStatusId != null) {
            assignmentStatusId = Long.parseLong(valueassignmentStatusId);
        } else {
            throw new IllegalArgumentException("'assignmentStatusId' parameter is required");
        }
        String valuescheduledNotificationId = request.getQueryString("scheduledNotificationId");
        Long scheduledNotificationId;
        if (valuescheduledNotificationId != null) {
            scheduledNotificationId = Long.parseLong(valuescheduledNotificationId);
        } else {
            scheduledNotificationId = null;
        }
        String valuetoDo = request.getQueryString("toDo");
        String toDo;
        if (valuetoDo != null) {
            toDo = valuetoDo;
        } else {
            toDo = null;
        }
        String valueconnection = request.getQueryString("connection");
        String connection;
        if (valueconnection != null) {
            connection = valueconnection;
        } else {
            connection = null;
        }
        String valuemethod = request.getQueryString("method");
        String method;
        if (valuemethod != null) {
            method = valuemethod;
        } else {
            method = null;
        }
        String valuestatus = request.getQueryString("status");
        String status;
        if (valuestatus != null) {
            status = valuestatus;
        } else {
            status = null;
        }
        String valueclosure = request.getQueryString("closure");
        String closure;
        if (valueclosure != null) {
            closure = valueclosure;
        } else {
            closure = null;
        }
        String valuemessage = request.getQueryString("message");
        String message;
        if (valuemessage != null) {
            message = valuemessage;
        } else {
            message = null;
        }
        String valuefollowUp = request.getQueryString("followUp");
        Long followUp;
        if (valuefollowUp != null) {
            followUp = Long.parseLong(valuefollowUp);
        } else {
            followUp = null;
        }
        String valueactive = request.getQueryString("active");
        Boolean active;
        if (valueactive != null) {
            active = Boolean.valueOf(valueactive);
        } else {
            active = null;
        }
        return imp.assignmentStatusUpdateHttp(request, version, accountId, assignmentStatusId, scheduledNotificationId, toDo, connection, method, status, closure, message, followUp, active);
    }

    @ApiAction
    public Result assignmentUpdate(Http.Request request, BigDecimal version) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueassignmentId = request.getQueryString("assignmentId");
        Long assignmentId;
        if (valueassignmentId != null) {
            assignmentId = Long.parseLong(valueassignmentId);
        } else {
            throw new IllegalArgumentException("'assignmentId' parameter is required");
        }
        String valuename = request.getQueryString("name");
        String name;
        if (valuename != null) {
            name = valuename;
        } else {
            name = null;
        }
        String valuedescription = request.getQueryString("description");
        String description;
        if (valuedescription != null) {
            description = valuedescription;
        } else {
            description = null;
        }
        String valueassigneeAccountId = request.getQueryString("assigneeAccountId");
        Long assigneeAccountId;
        if (valueassigneeAccountId != null) {
            assigneeAccountId = Long.parseLong(valueassigneeAccountId);
        } else {
            assigneeAccountId = null;
        }
        String valueretailerLocationId = request.getQueryString("retailerLocationId");
        Long retailerLocationId;
        if (valueretailerLocationId != null) {
            retailerLocationId = Long.parseLong(valueretailerLocationId);
        } else {
            retailerLocationId = null;
        }
        String valuetags = request.getQueryString("tags");
        String tags;
        if (valuetags != null) {
            tags = valuetags;
        } else {
            tags = null;
        }
        String valueactive = request.getQueryString("active");
        Boolean active;
        if (valueactive != null) {
            active = Boolean.valueOf(valueactive);
        } else {
            active = null;
        }
        return imp.assignmentUpdateHttp(request, version, accountId, assignmentId, name, description, assigneeAccountId, retailerLocationId, tags, active);
    }

}

File: server-scala-play-server/app/api/AssignmentApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.AccountMiniResponse
import model.AssignmentResponse
import model.AssignmentStatusResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class AssignmentApiController @Inject()(cc: ControllerComponents, api: AssignmentApi) extends AbstractController(cc) {
  /**
    * GET /api/3.18/assignment/assignee/search?accountId=[value]&keyword=[value]
    */
  def assigmentAssigneeAccountSearch(): Action[AnyContent] = Action { request =>
    def executeApi(): List[AccountMiniResponse] = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val keyword = request.getQueryString("keyword")
        
      api.assigmentAssigneeAccountSearch(accountId, keyword)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/create?accountId=[value]&name=[value]&description=[value]&assigneeAccountId=[value]&retailerLocationId=[value]&tags=[value]&active=[value]
    */
  def assignmentCreate(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val name = request.getQueryString("name")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("name", "query string")
        }
        
      val description = request.getQueryString("description")
        
      val assigneeAccountId = request.getQueryString("assigneeAccountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assigneeAccountId", "query string")
        }
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val tags = request.getQueryString("tags")
        
      val active = request.getQueryString("active")
        .map(value => value.toBoolean)
        
      api.assignmentCreate(accountId, name, assigneeAccountId, description, retailerLocationId, tags, active)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/delete?accountId=[value]&assignmentId=[value]
    */
  def assignmentDelete(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentId = request.getQueryString("assignmentId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentId", "query string")
        }
        
      api.assignmentDelete(accountId, assignmentId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/assignment/get?accountId=[value]&assignmentId=[value]
    */
  def assignmentGet(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentId = request.getQueryString("assignmentId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentId", "query string")
        }
        
      api.assignmentGet(accountId, assignmentId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/assignment/search?accountId=[value]&creatorAccountId=[value]&assigneeAccountIds=[value]&retailerLocationIds=[value]&currentStatusType=[value]&keyword=[value]&sortField=[value]&descending=[value]&activeOnly=[value]&start=[value]&limit=[value]
    */
  def assignmentSearch(): Action[AnyContent] = Action { request =>
    def executeApi(): List[AssignmentResponse] = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val creatorAccountId = request.getQueryString("creatorAccountId")
        .map(value => value.toLong)
        
      val assigneeAccountIds = request.getQueryString("assigneeAccountIds")
        
      val retailerLocationIds = request.getQueryString("retailerLocationIds")
        
      val currentStatusType = request.getQueryString("currentStatusType")
        
      val keyword = request.getQueryString("keyword")
        
      val sortField = request.getQueryString("sortField")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("sortField", "query string")
        }
        
      val descending = request.getQueryString("descending")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("descending", "query string")
        }
        
      val activeOnly = request.getQueryString("activeOnly")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("activeOnly", "query string")
        }
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("start", "query string")
        }
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("limit", "query string")
        }
        
      api.assignmentSearch(accountId, sortField, descending, activeOnly, start, limit, creatorAccountId, assigneeAccountIds, retailerLocationIds, currentStatusType, keyword)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/status/create?accountId=[value]&assignmentId=[value]&scheduledNotificationId=[value]&toDo=[value]&connection=[value]&method=[value]&status=[value]&closure=[value]&message=[value]&followUp=[value]&active=[value]
    */
  def assignmentStatusCreate(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentStatusResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentId = request.getQueryString("assignmentId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentId", "query string")
        }
        
      val scheduledNotificationId = request.getQueryString("scheduledNotificationId")
        .map(value => value.toLong)
        
      val toDo = request.getQueryString("toDo")
        
      val connection = request.getQueryString("connection")
        
      val method = request.getQueryString("method")
        
      val status = request.getQueryString("status")
        
      val closure = request.getQueryString("closure")
        
      val message = request.getQueryString("message")
        
      val followUp = request.getQueryString("followUp")
        .map(value => value.toLong)
        
      val active = request.getQueryString("active")
        .map(value => value.toBoolean)
        
      api.assignmentStatusCreate(accountId, assignmentId, scheduledNotificationId, toDo, connection, method, status, closure, message, followUp, active)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/status/delete?accountId=[value]&assignmentStatusId=[value]
    */
  def assignmentStatusDelete(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentStatusId = request.getQueryString("assignmentStatusId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentStatusId", "query string")
        }
        
      api.assignmentStatusDelete(accountId, assignmentStatusId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/assignment/status/get?accountId=[value]&assignmentStatusId=[value]
    */
  def assignmentStatusGet(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentStatusResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentStatusId = request.getQueryString("assignmentStatusId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentStatusId", "query string")
        }
        
      api.assignmentStatusGet(accountId, assignmentStatusId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/3.18/assignment/status/search?accountId=[value]&assignmentId=[value]&creatorAccountId=[value]&assigneeAccountId=[value]&retailerLocationId=[value]&statusType=[value]&keyword=[value]&sortField=[value]&descending=[value]&activeOnly=[value]&start=[value]&limit=[value]
    */
  def assignmentStatusSearch(): Action[AnyContent] = Action { request =>
    def executeApi(): List[AssignmentStatusResponse] = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentId = request.getQueryString("assignmentId")
        .map(value => value.toLong)
        
      val creatorAccountId = request.getQueryString("creatorAccountId")
        .map(value => value.toLong)
        
      val assigneeAccountId = request.getQueryString("assigneeAccountId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val statusType = request.getQueryString("statusType")
        
      val keyword = request.getQueryString("keyword")
        
      val sortField = request.getQueryString("sortField")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("sortField", "query string")
        }
        
      val descending = request.getQueryString("descending")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("descending", "query string")
        }
        
      val activeOnly = request.getQueryString("activeOnly")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("activeOnly", "query string")
        }
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("start", "query string")
        }
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("limit", "query string")
        }
        
      api.assignmentStatusSearch(accountId, sortField, descending, activeOnly, start, limit, assignmentId, creatorAccountId, assigneeAccountId, retailerLocationId, statusType, keyword)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/status/update?accountId=[value]&assignmentStatusId=[value]&scheduledNotificationId=[value]&toDo=[value]&connection=[value]&method=[value]&status=[value]&closure=[value]&message=[value]&followUp=[value]&active=[value]
    */
  def assignmentStatusUpdate(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentStatusResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentStatusId = request.getQueryString("assignmentStatusId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentStatusId", "query string")
        }
        
      val scheduledNotificationId = request.getQueryString("scheduledNotificationId")
        .map(value => value.toLong)
        
      val toDo = request.getQueryString("toDo")
        
      val connection = request.getQueryString("connection")
        
      val method = request.getQueryString("method")
        
      val status = request.getQueryString("status")
        
      val closure = request.getQueryString("closure")
        
      val message = request.getQueryString("message")
        
      val followUp = request.getQueryString("followUp")
        .map(value => value.toLong)
        
      val active = request.getQueryString("active")
        .map(value => value.toBoolean)
        
      api.assignmentStatusUpdate(accountId, assignmentStatusId, scheduledNotificationId, toDo, connection, method, status, closure, message, followUp, active)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * POST /api/3.18/assignment/update?accountId=[value]&assignmentId=[value]&name=[value]&description=[value]&assigneeAccountId=[value]&retailerLocationId=[value]&tags=[value]&active=[value]
    */
  def assignmentUpdate(): Action[AnyContent] = Action { request =>
    def executeApi(): AssignmentResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val assignmentId = request.getQueryString("assignmentId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("assignmentId", "query string")
        }
        
      val name = request.getQueryString("name")
        
      val description = request.getQueryString("description")
        
      val assigneeAccountId = request.getQueryString("assigneeAccountId")
        .map(value => value.toLong)
        
      val retailerLocationId = request.getQueryString("retailerLocationId")
        .map(value => value.toLong)
        
      val tags = request.getQueryString("tags")
        
      val active = request.getQueryString("active")
        .map(value => value.toBoolean)
        
      api.assignmentUpdate(accountId, assignmentId, name, description, assigneeAccountId, retailerLocationId, tags, active)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}
